src/serialization/sb3.js

```diff
@@ -271,6 +271,19 @@ const compressInputTree = function (block, blocks) {
     return block;
 };
 
+/**
+ * Get non-core extension ID for a given sb3 opcode.
+ * @param {!string} opcode The opcode to examine for extension.
+ * @return {?string} The extension ID, if it exists/is not a core extension.
+ */
+const getExtensionIdForOpcode = function (opcode) {
```

Comment:
I think this should probably be a utility function instead of living in `sb3.js`. I think it's used in a bunch of places including `sb2.js`, so it would be useful to abstract it out further.

Reply:
Is the list of `CORE_EXTENSIONS` necessarily the same between sb3/sb2? Or should the utility consume that list?

Reply:
Given the CORE_EXTENSIONS stuff, I think this probably should live in sb3. Either that or we need to change how we have 3 declarations of a constant called "CORE_EXTENSIONS", some of which are different.